dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpToolCallParams.java

```diff
@@ -82,19 +87,35 @@ private Object[] parseArguments(JsonObject arguments2, Jsonb jsonb) {
             if (argMetadata != null) {
                 String json = jsonb.toJson(argValue);
                 results[argMetadata.index()] = jsonb.fromJson(json, argMetadata.type());
-
             }
             argsProcessed.add(argName);
         }
 
-        if (!argsProcessed.equals(metadata.arguments().keySet())) {
-            List<String> data = generateArgumentMismatchData(argsProcessed, metadata.arguments().keySet());
-            throw new JSONRPCException(JSONRPCErrorCode.INVALID_PARAMS, data);
+        validateProcessedArgs(argsProcessed, arguments.keySet());
+
+        if (specialArguments.isEmpty()) {
+            return results;
+        }
+
+        HashSet<String> specialArgsProcessed = new HashSet<>();
+        for (var entry : specialArguments.entrySet()) {
+            SpecialArgumentMetadata specialArgsMetadata = entry.getValue();
+            results[specialArgsMetadata.index()] = specialArgsMetadata.type();
```

Comment:
Confused by this - we're putting an instance of `SpecialArgumentType` into the arguments array?